src/background.js

```diff
@@ -289,6 +289,23 @@ function getRootDomain(url) {
 
 }
 
+function getFrameAncestorsURLAndCheckIfFacebook(frameAncestorsArray) {
+  if (!frameAncestorsArray || frameAncestorsArray.length === 0) {
+    // No frame ancestor return false
+    return false;
+  }
+
+  const appsFacebookURL = "https://apps.facebook.com";
+  const frameAncestorsURL = frameAncestorsArray[0].url;
+
+  if (!frameAncestorsURL.startsWith(appsFacebookURL)) {
+    // Only allow frame ancestors that originate from apps.facebook.com
+    return false;
+  }
+
+  return isFacebookURL(frameAncestorsURL);
```

Comment:
Does this need to call `isFacebookURL` ? Didn't the check above determine that the `frameAncestorsURL` `startsWith` `https://apps.facebook.com` already?

Reply:
Great call. Revised!